### PR TITLE
APIv4 - Add function to get api class name

### PR DESCRIPTION
**Pull request by colemanw**
Overview
----------------------------------------
This PR does some APIv4 cleanup in preparation for adding a Case API.

Technical Details
----------------------------------------
Adding the "Case" entity is a problem because it's a reserved php keyword.
So we need to be able to have entity names that don't match their classname.
This adds a function to get api class names without guessing from the entity name.

### CODE REVIEW

File: Civi/API/Request.php [hunk 1]
```diff
@@ -48,7 +50,7 @@ public static function create(string $entity, string $action, array $params) {
           $apiRequest = \Civi\Api4\CustomValue::$action(substr($entity, 7));
         }
         else {
-          $callable = ["\\Civi\\Api4\\$entity", $action];
+          $callable = [CoreUtil::getApiClass($entity), $action];
```
Comment (totten) [inline Civi/API/Request.php:53]: Oh yes please!

File: Civi/Api4/Action/Entity/Get.php [hunk 2]
```diff
@@ -57,12 +59,14 @@ protected function getRecords() {
         foreach (glob("$dir/*.php") as $file) {
           $matches = [];
           preg_match('/(\w*)\.php$/', $file, $matches);
-          $entity = '\Civi\Api4\\' . $matches[1];
+          // Because "Case" is a reserved php keyword
+          $entityName = $matches[1] === 'CiviCase' ? 'Case' : $matches[1];
+          $className = '\Civi\Api4\\' . $matches[1];
```
Comment (totten) [inline Civi/Api4/Action/Entity/Get.php:64]: Any particular reason why this one still uses direct string manipulation?

Comment (colemanw) [reply Civi/Api4/Action/Entity/Get.php:64]: This is a very low-level function, and it's doing the reverse of what the new utility function does. But hang on I might have a more elegant solution...

Comment (colemanw) [reply Civi/Api4/Action/Entity/Get.php:62]: Ok, pushed up a better version without the specific reference to CiviCase.

**Comment by colemanw** [comment]
retest this please

**Comment by eileenmcnaughton** [comment]
Looks good - test cover is thorough here

**Pull request merged**